version/src/lib.rs

```diff
@@ -42,16 +43,36 @@ pub mod error {
         UnsupportedScaleEncode,
         /// JSON (de)serialization issue
         #[cfg(feature = "json")]
-        Serde(#[cfg_attr(feature = "std", source)] serde_json::Error),
+        Serde,
         /// Parity SCALE (de)serialization issue
         #[cfg(feature = "scale")]
-        ParityScale(#[cfg_attr(feature = "std", source)] parity_scale_codec::Error),
+        ParityScale,
         /// Problem with parsing integers
-        ParseInt(#[cfg_attr(feature = "std", source)] core::num::ParseIntError),
+        ParseInt,
         /// Input version unsupported
         UnsupportedVersion(UnsupportedVersion),
     }
 
+    #[cfg(feature = "json")]
+    impl From<serde_json::Error> for Error {
+        fn from(_: serde_json::Error) -> Self {
+            Self::Serde
+        }
+    }
+
+    #[cfg(feature = "scale")]
+    impl From<parity_scale_codec::Error> for Error {
+        fn from(_: parity_scale_codec::Error) -> Self {
+            Self::ParityScale
+        }
+    }
+
+    impl From<core::num::ParseIntError> for Error {
+        fn from(_: core::num::ParseIntError) -> Self {
+            Self::ParseInt
+        }
+    }
+
     impl fmt::Display for Error {
         fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
             let msg = match *self {
```

Comment:
No need for dereferencing here too. I think clippy needs a feature to catch that

Reply:
Thank you for pointing out. You can include this fix in your PR

Reply:
Yeah, will do it